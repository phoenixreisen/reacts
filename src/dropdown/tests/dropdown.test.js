/**
 * @jest-environment jsdom
 */
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Dropdown from '../dropdown.r';
import React from 'react';

describe('Dropdown should', () => {

    test('render correctly', () => {
        const component = renderer.create(
            <Dropdown title="Awesome Dropdown" icon="fa-user">
                <div>Element 1</div>
                <div>Element 2</div>
                <div>Element 3</div>
            </Dropdown>
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('contain all given children', () => {
        const component = render(
            <Dropdown>
                <div>Element</div>
                <div>Element</div>
                <div>Element</div>
                <div>Element</div>
            </Dropdown>
        );
        expect(component.getAllByText('Element')).toHaveLength(4);
    });

    test('toggle open status correctly', () => {
        const component = render(
            <Dropdown title="Awesome Dropdown">
                <div>Element</div>
            </Dropdown>
        );
        expect(document.querySelector('.dropdown--open')).toBeNull();
        fireEvent.click(component.getByText('Awesome Dropdown'));
        expect(document.querySelector('.dropdown--open')).not.toBeNull();
    });

    test('toggle open status on esc', () => {
        const component = render(
            <Dropdown title="Dropdown">
                <div>Element</div>
            </Dropdown>
        );
        expect(document.querySelector('.dropdown--open')).toBeNull();
        fireEvent.click(component.getByText('Dropdown'));
        expect(document.querySelector('.dropdown--open')).not.toBeNull();
        fireEvent.keyDown(component.getByText('Dropdown'), {keyCode: 27, key: 'Escape'});
        expect(document.querySelector('.dropdown--open')).toBeNull();
    });
});