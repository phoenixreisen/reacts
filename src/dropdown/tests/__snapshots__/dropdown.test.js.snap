// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
<article
  className="dropdown  "
>
  <a
    href="#"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fas fa-user mr1"
    />
     
    Awesome Dropdown
  </a>
  <div
    className="dropdown-items"
  >
    <div
      className="dropdown-item dropdown-item-0"
      onClick={[Function]}
    >
      <div>
        Element 1
      </div>
    </div>
    <div
      className="dropdown-item dropdown-item-1"
      onClick={[Function]}
    >
      <div>
        Element 2
      </div>
    </div>
    <div
      className="dropdown-item dropdown-item-2"
      onClick={[Function]}
    >
      <div>
        Element 3
      </div>
    </div>
  </div>
</article>
`;
